demos/espressif/esp32_devkitc_esp_wrover_kit/make/Makefile

```diff
@@ -2,18 +2,25 @@
 # This is a project Makefile. It is assumed the directory this Makefile resides in is a
 # project subdirectory.
 #
+# To use your own main.c, clone this demo directory and modify the
+# AMAZON_FREERTOS_PATH below or pass it to make on the command line
 
 PROJECT_NAME := aws_demos
 
 ifndef AMAZON_FREERTOS_PATH
 export AMAZON_FREERTOS_PATH := $(CURDIR)/../../../..
 endif
 
+# Amazon FreeRTOS relies on the bundled ESP-IDF. If you are sure you need to override the ESP-IDF,
+# also set the variable CUSTOM_IDF_PATH
 ifndef IDF_PATH
-export IDF_PATH := $(AMAZON_FREERTOS_PATH)/lib/third_party/mcu_vendor/espressif/esp-idf
+	export IDF_PATH := $(AMAZON_FREERTOS_PATH)/lib/third_party/mcu_vendor/espressif/esp-idf
+else
+	ifndef CUSTOM_IDF_PATH
+	$(error "IDF path is set. Amazon FreeRTOS expects a specific ESP-IDF. Pleae unset IDF_PATH or if you are sure, set CUSTOM_IDF_PATH and rerun.")
```

Comment:
Are we sure we want to allow this? It would be nice if we didn't need to use a specific version, but anecdotally I've gotten all kinds of strange build errors from accidentally using a different version of ESP-IDF.

Reply:
Right now there is no check for a user-provided IDF PATH. See related issues: #375 , #497, #338, #87. This check tightens things up a little but still provides an escape route

Reply:
Is anyone asking for an escape route, though? It looks like those issues were all resolved by un-setting the IDF_PATH variable. Should we go out of our way to include this behavior if it is basically just an option which lets you voluntarily break your build?

People can still modify these Makefiles if they want to tinker with the build system themselves, but I'm not sure if we should make it sound like something that we support.